src/graph_compute/update_function.h

```diff
@@ -58,7 +65,16 @@ typedef struct vertex_t vertex_t;
 //  a sched_t datatype which includes whatever per-vertex data
 //  they need to perform their scheduling (e.g., priority, satisfied, dependencies etc.)
 #if PAGERANK
-  inline static void fillInNodeData(vertex_t * nodes, const vid_t cntNodes) {
+  inline static uint64_t hashOfVertexData(data_t * vertex) {
+    uint64_t result = 0;
+    memcpy(static_cast<void *>(&result),
```

Comment:
The problem is that there is no way to know if `sizeof(pagerank_t)` is no bigger than `sizeof(uint64_t)` other than by reading all of the source code, and that presents an excellent opportunity to shoot oneself in the foot. Since this isn't central to our code, I'd be happy with an assertion like `assert(sizeof(pagerank_t) <= sizeof(uint64_t))` before the memcpy.

Also, I don't think you need the static casts -- there is an implicit cast from `anything *` to `void *`.